source/Irrlicht/CBAWMeshFileLoader.h

```diff
@@ -20,16 +20,18 @@ namespace irr { namespace scene
 class CBAWMeshFileLoader : public IMeshLoader
 {
 private:
-	struct SPair
+	struct SBlobData
 	{
-		core::BlobHeaderV1* header;
-		void* blob;
+		const core::BlobHeaderV1* header;
+		const void* blob;
+		bool wasCreated;
 	};
 
 	struct SContext
 	{
 		io::path filePath;
-		std::map<uint64_t, SPair> blobs;
+		std::map<uint64_t, SBlobData> blobs;
+		std::map<uint64_t, IReferenceCounted*> createdObjects;
```

Comment:
Look at my SBlobData comment, you can ditch this in favor of an `IReferenceCounted* createdObj` in SBlobData.